addons/crm/models/crm_team.py

```diff
@@ -463,7 +473,15 @@ def _allocate_leads(self, creation_delta_days=7):
 
             # assign + deduplicate and concatenate results in teams_data to keep some history
             candidate_lead = teams_data[team]["leads"][0]
-            assign_res = team._allocate_leads_deduplicate(candidate_lead, duplicates_cache=duplicates_lead_cache)
+            try:
+                assign_res = team._allocate_leads_deduplicate(candidate_lead, duplicates_cache=duplicates_lead_cache)
+            except Exception as e:
+                if auto_commit:
+                    _log_team_data(teams_data, global_data)
+                    _logger.error("[Assign Lead] Crash while assigning lead %s (ID: %d) to team %s (ID: %d). Error: %s", candidate_lead.name, candidate_lead.id, team.name, team.id, str(e))
```

Comment:
We can keep this log on error however, it's good to know which one caused the crash so it can be investigated.
```suggestion
                    _logger.error("[Assign Lead] Error while assigning lead %s (ID: %d) to team %s (ID: %d). Error: %s", candidate_lead.name, candidate_lead.id, team.name, team.id, str(e))
```